src/boot.c

```diff
@@ -294,6 +295,20 @@ static void check_backup_boot(DOS_FS * fs, struct boot_sector *b, unsigned int l
 	    return;
     }
 
+    if (boot_sector && fs->backupboot_start &&
```

Comment:
Is this check correct? Because by default is boot_sector zero.

Reply:
Yeah, that's only if the backup boot sector is being used (i.e. `boot_sector != 0`).

Reply:
So what is this code doing? And why is this check needed? Because it is completely unclear / unreadable what is going here...